File: src/test/resources/collections/Map2.java
package collections;

import java.util.Map;

public class Map2 {
	static void test() {
		java.util.Map<String, String> map = new java.util.HashMap<>();
		if (map.containsKey("foo")) {
			Object value = map.get("foo");
		}
		for (Object value : map.values()) {
		}
		for (Object key : map.keySet()) {
		}
		Object removed = map.remove("foo");
		map.put("foo", "bar");
		String evicted = map.put("foo", "bar2");
	}

	static String putEvict(Map<String, String> map) {
	    return map.put("a", "b");
    }

	static java.util.Map genericSortedMap() {
		return new java.util.TreeMap<String, String>();
	}
	
	static java.util.Map sortedMap() {
		return new java.util.TreeMap();
	}
}

File: src/test/resources/collections/Map2.cs
namespace collections
{
	public class Map2
	{
		internal static void test()
		{
			System.Collections.Generic.IDictionary<string, string> map = new System.Collections.Generic.Dictionary
				<string, string>();
			if (map.ContainsKey("foo"))
			{
				object value = map.GetOrNull("foo");
			}
			foreach (object value in map.Values)
			{
			}
			foreach (object key in map.Keys)
			{
			}
			object removed = Sharpen.Collections.Remove(map, "foo");
			Sharpen.Collections.Put(map, "foo", "bar");
			string evicted = Sharpen.Collections.Put(map, "foo", "bar2");
		}

		internal static string putEvict(System.Collections.Generic.IDictionary<string, string
			> map)
		{
			return Sharpen.Collections.Put(map, "a", "b");
		}

		internal static System.Collections.IDictionary genericSortedMap()
		{
			return new System.Collections.Generic.SortedDictionary<string, string>();
		}

		internal static System.Collections.IDictionary sortedMap()
		{
			return new System.Collections.Generic.SortedDictionary<object, object>();
		}
	}
}